### ISSUE TITLE
How can I get the actual fn output as well as derivative of parameters when using `grad`?

### ISSUE THREAD

**Issue by freckletonj** [HIPS/autograd#496]
So far, I've been doing this:

```python
def f(params, inputs, target):
    ...
    return loss, output

f_loss = lambda params, inputs, target: f(params, inputs, target)[1]
g = grad(f_loss)
```

But I have a function where, when `f` is ran, I want the derivative of `params` along with the `output` and `loss` that the execution of `f` produced.

I tried this:

```python
def new_f(params, ...):
    loss, o = f(params, ...)
    d_params = grad(lambda params: loss)(params)
    return d_params, loss, o
```

But, by this point it's too late to get a derivative of loss. So, it returns the "gradient" of the now constant `loss`, which is all 0s.

Thanks for this amazing library btw! :)

**Comment by j-towns** [comment]
Try `value_and_grad`! You can import it with
```python
from autograd import value_and_grad
```

**Comment by freckletonj** [comment]
Thanks! In searching for `value_and_grad` I also found `grad_and_aux` which is useful. 

Note to timetravelers, these return `ArrayBox`, and you can get your normal `ndarray` using `thing._value`.
